### PR TITLE
PYTHON-861: query: make sure the retry_policy is passed to BoundStatements

### PR DESCRIPTION
**Pull request by iksaif**
Before this patch all prepared statements would end up with
the default RetryPolicy.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,3 +1,10 @@
+3.13.0
+======
+
+Bug Fixes
+---------
+* Support retry_policy in PreparedStatement
```
Comment (bjmb) [inline CHANGELOG.rst]: Can you add here the number of the ticket, like
Support retry_policy in PreparedStatement (PYTHON-861)

Comment (iksaif) [reply CHANGELOG.rst]: done

**Comment by iksaif** [comment]
Yes, the goal is to avoid explicitly binding prepared statements (since this happens implicitly in `execute`)

**Comment by bjmb** [comment]
Thanks @iksaif!  Merging

**Pull request merged**